src/components/DataViewer/DataViewer.tsx

```diff
@@ -0,0 +1,108 @@
+"use client";
+
+import React from "react";
+import styled from "styled-components";
+import Modal from "react-modal";
+
+interface DataViewerProps {
+    data: { [key: string]: string | number | null };
+    title: string;
+    isOpen: boolean;
+    onRequestClose: () => void;
+}
+
+const Header = styled.h3`
+    display: flex;
+    justify-content: space-between;
+    background-color: #eeeeee;
+    margin-bottom: 1rem;
+    margin-top: 0;
+    padding: 0.5rem;
+`;
+
+const EachItem = styled.div`
+    display: flex;
+    flex-direction: column;
+    padding-bottom: 1rem;
+`;
+
+const ContentWrapper = styled.div`
+    overflow: scroll;
+    height: calc(100% - 5rem);
+    margin: 1.5rem;
+`;
+
+const ModalWrapper = styled.div`
+    width: 100%;
+    height: 100%;
+    overflow: hidden;
+`;
+
+const Key = styled.div`
+    color: grey;
+    font-size: small;
+`;
+
+const Value = styled.div`
+    font-size: medium;
+`;
+
+const CloseButton = styled.button`
+    border: 0;
+    background-color: red;
+    color: white;
+`;
+
+const DataViewer: React.FC<DataViewerProps> = (props) => {
+    const ClearButton: React.FC = () => {
+        return <CloseButton onClick={closeModel}>X</CloseButton>;
+    };
+
+    const JSONContent = Object.entries(props.data).map(([key, value]) => {
+        return (
+            <EachItem key={key}>
+                <Key>{key.toUpperCase().replace("_", " ")}</Key>
+                <Value>{value ?? ""}</Value>
+            </EachItem>
+        );
+    });
+
+    const closeModel: () => void = () => {
+        props.onRequestClose();
+    };
+
+    if (!props.isOpen) {
+        return <></>;
+    }
+
+    return (
+        <Modal
+            isOpen={true}
+            onRequestClose={closeModel}
+            ariaHideApp={false}
+            contentLabel="JSON Modal"
+            style={{
+                content: {
+                    padding: "0",
+                    border: "0",
+                    width: "80%",
+                    height: "80%",
+                    boxShadow: "0px 2px 10px rgb(200, 200, 200)",
+                    top: "50%",
+                    left: "50%",
+                    transform: "translate(-50%, -50%)",
+                },
+            }}
```

Comment:
Ask Tom, but you might want to replace this using styled-components by creating a custom `<StyledModal>`  component within this file. This might not be needed, and may not be worth the effort, but it's something to ask.

Reply:
yep, wherever possible (and it is possible here!) I don't want `style=` set, and I would like `styled-components` to be used.
Potentially ask @SamYap-Softwire / @Var1377 for help, they both did some styled component stuff with modals!